frontend/apps/reader/modules/readerview.lua

```diff
@@ -695,7 +695,7 @@ function ReaderView:recalculate()
             -- start from right of page_area
             self.visible_area.x = self.page_area.x + self.page_area.w - self.visible_area.w
         end
-        if self.ui.zooming.zoom_bottom_to_top then
+        if self.document.configurable.zoom_direction >= 2 and self.document.configurable.zoom_direction <= 5 then -- zoom_bottom_to_top
```

Comment:
Ok, this doesn't look too bad :)
A bit sad to put here some logic (about `configurable.zoom_direction`) that should better be only in ReaderZooming (I guess you do that because ReaderZooming may not yet be available), but it looks like we already do the same kind of thing just above for `configurable.writing_direction`, so the sadness feels a bit consistent and we'll soon forget about it :)